src/main/java/org/fcrepo/spec/testsuite/versioning/AbstractVersioningTest.java

```diff
@@ -171,14 +174,44 @@ protected String createMemento(final String originalResourceUri, final String me
         final Headers headers = new Headers(new Header("Content-Type", contentType),
                                             new Header(MEMENTO_DATETIME_HEADER, mementoDateTime));
 
-        final Response timeMapResponse = doPost(timeMapURI.toString(),headers, body);
+        final Response timeMapResponse = doPost(timeMapURI.toString(), headers, body);
         return getLocation(timeMapResponse);
     }
 
+    protected URI getTimeGateUri(final Response response) {
+        return getLinksOfRelTypeAsUris(response, "timegate").findFirst().get();
+    }
+
     protected String createMemento(final String originalResourceUri) {
         final Response response = doGet(originalResourceUri);
         final URI timeMapURI = getTimeMapUri(response);
-        final Response timeMapResponse = doPost(timeMapURI.toString());
-        return getLocation(timeMapResponse);
+        if (hasHeaderValueInMultiValueHeader("Allow", "Post", doGet(timeMapURI.toString()))) {
+            //if POST allowed (client-managed versioning)
+            return getLocation(doPost(timeMapURI.toString()));
+        } else {
+            // otherwise create a memento by altering the original resource and retrieving the most recent memento
+            //if ldp-rs
+
+            if (getLinksOfRelType(response, "type").anyMatch(link -> link.equals(RDF_SOURCE_LINK_HEADER))) {
```

Comment:
`link.equals(RDF_SOURCE_LINK_HEADER)` will never be true since `link` is of type Link and the other is of type String.